### PR TITLE
Fix Python print

### PR DESCRIPTION
**Pull request by scls19fr**
### Requirements

Print is now a function since Python 3.0

### Description of the Change

```
2to3 -f print -n -l -w .
```

and some manual fixes

### Benefits

Should works both with Python 2.7 and 3.x

### Possible Drawbacks

Maybe additional spaces for 

```
print x, y
```

which are changed automatically to

```
print(x, y)
```

instead of manually fixing to

```
print("%s %s" % (x, y))
```

other possible drawback:

this fix will remove possible compatibility with very old Python 2.x version
(ie version which doesn't support print as a function)

other possible drawback:

with Python 2.7

```
print()
```

prints

```
()\n
```

but with Python 3.x

```
\n
```

so changing manually to

```
print("")
```

but I may have miss some

### Verification Process

Run "some" scripts (but not all)

### Applicable Issues

Related #1612

### CODE REVIEW

**Comment by ibell** [comment]
I very much appreciate the fixing of this - can you also add
```
from __future__ import print_function
```
to the top of all scripts that you modified?  That will fix the print function in python 2.7.  Until it reaches end-of-life, python 2.7 is still a thing.

**Comment by scls19fr** [comment]
Sorry but I'm not sure that I can do this automatically as there are some `.py` files that can have such an import.
I must admit that I'm not a `grep` / `sed` expert.
Anyway I don't think that printing sometimes not perfectly things (only with Py2.7) is something really annoying compared to the fact that all modified scripts are (were) broken with Py3.x.

**Comment by ibell** [comment]
Fair enough - most of the scripts are not actually part of CoolProp anyhow,
but are/were used for development purposes.  And most of that has
historically at least been done in 2.7, but I'm finally leaving that all
behind.

**Comment by scls19fr** [comment]
I think it's more "future" oriented to watch at diff and fix manual print calls that need to be fixed.

File: dev/Tickets/524.py [hunk 1]
```diff
@@ -2,42 +2,42 @@
 from CoolProp.CoolProp import PropsSI
 from CoolProp.CoolProp import set_reference_state
 
-print "CoolProp version",CoolProp.__version__
-print "CoolProp revision",CoolProp.__gitrevision__
+print("CoolProp version",CoolProp.__version__)
```
Comment (scls19fr) [inline dev/Tickets/524.py]: This print call for example need to be fixed manually

File: dev/Tickets/524.py [hunk 2]
```diff
@@ -2,42 +2,42 @@
...
+print("CoolProp revision",CoolProp.__gitrevision__)
```
Comment (scls19fr) [inline dev/Tickets/524.py]: also this one

**Comment by ibell** [comment]
I'm inclined to say we merge this PR, but I wish you had only changed the
print statement to a print function and made no other changes; that's a
change that I would whole-heartedly endorse.  I'm not so sure about all
these modified string insertion updates, I wish they had been left
untouched since they are valid Python code.

File: dev/TTSE/speed_test_TTSE_BICUBIC.py [hunk 3]
```diff
@@ -30,4 +30,4 @@ def time_check(N, h, p, TTSE = False, mode = 'TTSE'):
     time_TTSE, value_TTSE = time_check(N, h, p, TTSE = True)
     time_BICUBIC, value_BICUBIC = time_check(N, h, p, TTSE = True, mode='BICUBIC')
 
-    print k, h, p, (value_TTSE/value_EOS-1.0)*100, (value_BICUBIC/value_EOS-1.0)*100, time_EOS/time_TTSE, time_EOS/time_BICUBIC
+    print(k, h, p, (value_TTSE/value_EOS-1.0)*100, (value_BICUBIC/value_EOS-1.0)*100, time_EOS/time_TTSE, time_EOS/time_BICUBIC)
```
Comment (scls19fr) [inline dev/TTSE/speed_test_TTSE_BICUBIC.py]: This one need to be fixed

**Comment by thorade** [comment]
This looks all good to me, but I only looked at it, did not run any tests or similar.
I think the from future imports can be added using the futurize script (it is based on 2to3).

File: dev/pseudo-pure/fit_pseudo-pure_eos.py [hunk 4]
```diff
@@ -358,7 +358,7 @@ def OBJECTIVE(self, N):
         residuals = np.r_[(PPF.p/self.p-1),(PPF.cv/self.cv-1),(PPF.cp/self.cp-1)]#,(PPF.w**2/self.speed_sound**2-1)]
         RMS = np.sqrt(np.mean(np.power(residuals, 2)))
 
-        print 'RMS:',RMS*100, '% Max',np.max(np.abs(residuals))*100,'%'
+        print('RMS: %s %% Max %s %' % (RMS*100, np.max(np.abs(residuals))*100))
```
Comment (scls19fr) [inline dev/pseudo-pure/fit_pseudo-pure_eos.py]: Forget to escape second percent

File: dev/genetic.py [hunk 5]
```diff
@@ -287,8 +287,8 @@ def run(self):
 
         self.fitness(samples[0])
 
-        print self.value
-        print '// Max error is ',samples[0].max_abserror,'% between',np.min(self.T),'and',np.max(self.T),'K'
+        print(self.value)
+        print('// Max error is ',samples[0].max_abserror,'% between',np.min(self.T),'and',np.max(self.T),'K')
```
Comment (scls19fr) [inline dev/genetic.py]: an other one!

File: dev/scripts/check_mole_masses.py [hunk 6]
```diff
@@ -10,4 +10,4 @@
         m += masses[el]*int(c)/1000.0
     err = m/CoolProp.CoolProp.PropsSI(fluid,"M")-1
     if abs(err) > 1e-3:
-        print(fluid, m/CoolProp.CoolProp.PropsSI(fluid,"M")-1)
+        print((fluid, m/CoolProp.CoolProp.PropsSI(fluid,"M")-1))
```
Comment (scls19fr) [inline dev/scripts/check_mole_masses.py]: should be fixed also

File: dev/scripts/derivative_tester.py [hunk 7]
```diff
@@ -39,38 +39,38 @@ def f(index, dx):
 rho = 1.5
 H = Props('H','T',T,'D',rho,fluid)
 P = Props('P','T',T,'D',rho,fluid)
-print T,rho,H,P
+print(T,rho,H,P)
```
Comment (scls19fr) [inline dev/scripts/derivative_tester.py]: same here

File: dev/scripts/derivative_tester.py [hunk 8]
```diff
@@ -83,137 +83,137 @@ def f(dx):
     return DerivTerms('dphir_dDelta',T,rho,fluid)
 
 
-print 'd2phir_dDelta2'
-print (f(-2*dx)-8*f(-dx)+8*f(dx)-f(2*dx))/(12*dx)
-print DerivTerms('d2phir_dDelta2',T,rho,fluid)
+print('d2phir_dDelta2')
+print((f(-2*dx)-8*f(-dx)+8*f(dx)-f(2*dx))/(12*dx))
```
Comment (scls19fr) [inline dev/scripts/derivative_tester.py]: not sure about what 2to3 did here

Comment (scls19fr) [reply dev/scripts/derivative_tester.py]: that's ok sorry for noise

File: dev/scripts/fit_ancillary_ODRPACK.py [hunk 9]
```diff
@@ -36,7 +36,7 @@ def saturation_density(Ref, ClassName, form = 'A', LV = 'L', perc_error_allowed
         pc = Props(Ref,'pcrit')
         rhoc = Props(Ref,'rhocrit')
         Tmin = Props(Ref,'Tmin')
-        print Ref,Tmin,Props(Ref,'Ttriple')
+        print(Ref,Tmin,Props(Ref,'Ttriple'))
```
Comment (scls19fr) [inline dev/scripts/fit_ancillary_ODRPACK.py]: to be fixed

File: dev/scripts/fit_ancillary_ODRPACK.py [hunk 10]
```diff
@@ -86,7 +86,7 @@ def saturation_density(Ref, ClassName, form = 'A', LV = 'L', perc_error_allowed
 
     max_abserror = 0
     while len(n) > 3:
-        print max_abserror, len(n)
+        print(max_abserror, len(n))
```
Comment (scls19fr) [inline dev/scripts/fit_ancillary_ODRPACK.py]: same

**Comment by scls19fr** [comment]
This is ready on my side! (I think)

File: wrappers/Python/CoolProp/__init__.py [hunk 11]
```diff
@@ -10,7 +10,7 @@
         os.remove(constants.__file__)
         print("constants shared library has been removed.  Please restart your python code")
     except:
-        print("Unable to remove" + constants.__file__ + ". Please manually remove it")
```
Comment (ibell) [inline wrappers/Python/CoolProp/__init__.py]: Can you please revert the changes in this file?

Comment (scls19fr) [reply wrappers/Python/CoolProp/__init__.py]: What's wrong here?

**Comment by ibell** [comment]
There was nothing wrong with the original, so I'd prefer to not touch any
file that was fine to begin with.  Makes me nervous, and its an unnecessary
clutter to the diff.

**Comment by scls19fr** [comment]
I think the main problem is that CoolProp should have unit tests to run all these scripts.
Moreover there is so many differents way of using `print` in all these scripts!
We can find probably every way of doing... except f-strings (which are only for Python 3.6 and above).
- “Old-school” String Formatting with `%s`
- concatenation with `str` calls
- `str.format()`
- ...

**Comment by scls19fr** [comment]
File reverted

**Comment by ibell** [comment]
These scripts are mostly not used any more, so none of these are actually
terminal problems with CoolProp, and unit testing is a bit overkill as they
shouldn't actually run regularly.  But I agree that it would be good to fix
print function everywhere

**Comment by ibell** [comment]
Here we go!
